### PR TITLE
Map order errors to response

### PR DESCRIPTION
**Pull request by seallard**

### CODE REVIEW

File: cg/services/order_validation_service/workflows/tomte/response_mapper.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+)
+
+
+def create_validated_order_response(raw_order: dict, errors: ValidationErrors) -> dict:
+    wrap_fields(raw_order)
+    map_errors_to_order(raw_order, errors)
+    return raw_order
+
+
+def map_errors_to_order(order: dict, errors: ValidationErrors) -> None:
```
[the remark this answers is not in the document]
Comment (seallard) [reply cg/services/order_validation_service/workflows/tomte/response_mapper.py:15]: So far just Tomte!

File: cg/services/order_validation_service/workflows/tomte/response_mapper.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def map_order_errors(order: dict, errors: list[OrderError]) -> None:
+    for error in errors:
+        order[error.field]["errors"].append(error.message)
+
+
+def map_case_errors(order: dict, errors: list[CaseError]) -> None:
+    for error in errors:
+        order["cases"][error.case_index][error.field]["errors"].append(error.message)
```
Comment (islean) [inline cg/services/order_validation_service/workflows/tomte/response_mapper.py:28]: Not a beautiful line but might be hard to wrap in a more understandable way.

File: cg/services/order_validation_service/workflows/tomte/response_mapper.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def map_case_sample_errors(order: dict, errors: list[CaseSampleError]) -> None:
+    for error in errors:
+        case: dict = get_case(order, error.case_index)
+        sample: dict = get_sample(case, error.sample_index)
+        add_error_to_sample_field(sample, error)
```
Comment (islean) [inline cg/services/order_validation_service/workflows/tomte/response_mapper.py:35]: Named arguments

File: cg/services/order_validation_service/workflows/tomte/response_mapper.py (hunk 2 above)
Comment (islean) [reply cg/services/order_validation_service/workflows/tomte/response_mapper.py:28]: We might need to do some null checking for not provided fields though

**Review by islean — COMMENTED** [review]
Nice approach! I do think we need to consider cases where the dict does not have all the keys though.

File: cg/services/order_validation_service/workflows/tomte/response_mapper.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    CaseError,
+    CaseSampleError,
+    OrderError,
+    ValidationErrors,
+)
+
+
+def create_validated_order_response(raw_order: dict, errors: ValidationErrors) -> dict:
+    wrap_fields(raw_order)
+    map_errors_to_order(raw_order, errors)
```
Comment (islean) [inline cg/services/order_validation_service/workflows/tomte/response_mapper.py:11]: named argument

**Review by islean — APPROVED** [review]
Nice!

**Pull request merged**